### PR TITLE
HTBHF-1712 Resend code if a user changes their contact details used t…

### PR DESCRIPTION
**Pull request by dwybourn**
After entering a confirmation code, the user now has to confirm a new code if they change the contact details used to receive a code.
i.e. changing your phone number will cause a new code to be send, only if the original code was texted.

Acceptance tests to come in next PR.

### CODE REVIEW

File: src/web/routes/application/common/confirmation-code.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// reset confirmation code and reset state to in progress so that the user is not taken straight back to check details.
```
Comment (YetAnotherMatt) [inline src/web/routes/application/common/confirmation-code.js]: Shouldn't this comment be against the handleConfirmationCodeReset method?

Comment (dwybourn) [reply src/web/routes/application/common/confirmation-code.js]: Fixed.

Comment (YetAnotherMatt) [reply src/web/routes/application/common/confirmation-code.js]: it hasn't moved yet...

File: src/web/routes/application/common/confirmation-code.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const { CONFIRMATION_CODE_ENTERED_SESSION_PROPERTY } = require('./constants')
+const { stateMachine, states } = require('../common/state-machine')
+
+const handleConfirmationCodeReset = req => {
+  req.session[CONFIRMATION_CODE_ENTERED_SESSION_PROPERTY] = false
```
Comment (gethyn1) [inline src/web/routes/application/common/confirmation-code.js]: Not for this PR, but I think some of the higher level constants files could be split out into distinct modules (e.g. paths, session keys etc..).

File: src/web/routes/application/email-address/email-address.test.js [hunk 3]
```diff
@@ -1,6 +1,14 @@
 const test = require('tape')
+const sinon = require('sinon')
+const proxyquire = require('proxyquire')
 const { contentSummary } = require('./email-address')
 
+const { TEXT, EMAIL } = require('../common/constants')
+
+const handleConfirmationCodeReset = sinon.spy()
+
+const { behaviourForPost } = proxyquire('./email-address', { '../common/confirmation-code': { handleConfirmationCodeReset } })
```
Comment (gethyn1) [inline src/web/routes/application/email-address/email-address.test.js]: I know PRs are not really about formatting so not required, but I would maybe reformat this to multiple lines as its quite a lot of inline `{ ... }` so a bit tricky to determine the nesting.

File: src/web/views/enter-code.njk [hunk 4]
```diff
@@ -34,5 +34,5 @@
 
     {# TODO DW HTBHF-1712 update link to send new confirmationCode  #}
     <p class="govuk-body-m"><a href="/send-code" id="request-new-code">{{ requestANewCode }}</a> {{ ifItDoesNotArrive }}</p>
```
Comment (gethyn1) [inline src/web/views/enter-code.njk]: Is there a way of putting the link in the translation? This implementation assumes that all languages will use the same order of words.

File: src/web/views/enter-code.njk [hunk 5]
```diff
@@ -34,5 +34,5 @@
...
-s
+
```
Comment (gethyn1) [inline src/web/views/enter-code.njk]: I must have missed this in a previous PR - is there a way of putting the link in the translation? This implementation assumes that all languages will use the same order of words.

File: src/web/routes/application/common/confirmation-code.js (hunk 1 above)
Comment (dwybourn) [reply src/web/routes/application/common/confirmation-code.js]: Not sure what happened there, fixed now.

File: src/web/views/enter-code.njk (hunk 5 above)
Comment (dwybourn) [reply src/web/views/enter-code.njk]: Will do in a separate PR on it's own.

**Pull request merged**